shinken/objects/item.py

```diff
@@ -904,37 +917,14 @@ def index_item(self, item, name=None):
         self.name_to_item[name] = item
         return item
 
-    def remove_item(self, item):
-        """
-        Removes and unindexes an item from the `items` container.
-
-        :param item: The item to remove
-        """
-        name_property = getattr(self.__class__, "name_property", None)
-        if name_property:
-            self.unindex_item(item)
-        try:
-            del self.items[item.id]
-        except KeyError:
-            pass
-
-    def unindex_item(self, item, name=None):
-        """
-        Unindexes an item from the `items` container.
-
-        An optional `name` may be given to indicate the mame under which the
-        object was indexed.
-
+    def unindex_item(self, item):
+        """ Unindex an item from our name_to_item dict.
         :param item:    The item to unindex
-        :param name:    The name under which the item has been indexed.
         """
         name_property = getattr(self.__class__, "name_property", None)
```

Comment:
Actually i'm figuring that there is no name_property for a  service object but there is one for an host one. Strange.